handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1857,10 +1849,10 @@ private String selectApplicationProtocol(List<String> protocols,
         @Override
         public Certificate[] getPeerCertificates() throws SSLPeerUnverifiedException {
             synchronized (ReferenceCountedOpenSslEngine.this) {
-                if (peerCerts == null || peerCerts.length == 0) {
+                if (peerCerts == null || peerCerts.isEmpty()) {
                     throw new SSLPeerUnverifiedException("peer not verified");
                 }
-                return peerCerts;
+                return peerCerts.toArray(new Certificate[peerCerts.size()]);
```

Comment:
@Scottmitch it just makes the `init*` method a lot easier to use an ArrayList. That said I can change if you want.